src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -939,105 +935,149 @@ public Response getAssignedGroupsByGameboards(@Context final HttpServletRequest
-    public Response assignGameBoard(@Context final HttpServletRequest request,
-                                    final AssignmentDTO assignmentDTOFromClient) {
-
-        if (assignmentDTOFromClient.getGameboardId() == null || assignmentDTOFromClient.getGroupId() == null) {
-            return new SegueErrorResponse(Status.BAD_REQUEST, "A required field was missing. Must provide group and gameboard ids").toResponse();
-        }
-
+    @ApiOperation(value = "Create one or more new assignment(s).")
+    public Response assignGameBoards(@Context final HttpServletRequest request,
+                                    final List<AssignmentDTO> assignmentDTOsFromClient) {
         try {
             RegisteredUserDTO currentlyLoggedInUser = userManager.getCurrentRegisteredUser(request);
-            UserGroupDTO assigneeGroup = groupManager.getGroupById(assignmentDTOFromClient.getGroupId());
-
+            // Assert user is allowed to set assignments
             boolean userIsTeacherOrAbove = isUserTeacherOrAbove(userManager, currentlyLoggedInUser);
             boolean userIsStaff = isUserStaff(userManager, currentlyLoggedInUser);
-            boolean notesIsNullOrEmpty = assignmentDTOFromClient.getNotes() == null || (assignmentDTOFromClient.getNotes() != null && assignmentDTOFromClient.getNotes().isEmpty());
-            boolean notesIsTooLong = assignmentDTOFromClient.getNotes() != null && assignmentDTOFromClient.getNotes().length() > MAX_NOTE_CHAR_LENGTH;
 
             if (!userIsTeacherOrAbove) {
                 return new SegueErrorResponse(Status.FORBIDDEN, "You need a teacher account to create groups and set assignments!").toResponse();
             }
 
-            if (null == assigneeGroup) {
-                return new SegueErrorResponse(Status.BAD_REQUEST, "The group id specified does not exist.")
-                        .toResponse();
+            // Assert that there is at least one assignment, and that multiple assignments are only set by staff
+            if (assignmentDTOsFromClient.size() == 0) {
+                return new SegueErrorResponse(Status.BAD_REQUEST, "You need to specify at least one assignment to set.").toResponse();
+            } else if (!userIsStaff && assignmentDTOsFromClient.size() > 1) {
+                return new SegueErrorResponse(Status.FORBIDDEN, "You need a staff account to set assignments to more than one group at once!").toResponse();
             }
 
-            if (!GroupManager.isOwnerOrAdditionalManager(assigneeGroup, currentlyLoggedInUser.getId())
-                    && !isUserAnAdmin(userManager, currentlyLoggedInUser)) {
-                return new SegueErrorResponse(Status.FORBIDDEN,
-                        "You can only set assignments to groups you own or manage.").toResponse();
-            }
+            // Since all of the assignment notes in a single query are currently the same, we only have to do note size
+            // checks once, and we only need to fetch the gameboard once
```

Comment:
This seems unlikely to be the case generally, since there's no guarantee the request is coming from the frontend. I think the notes for each assignment need to be checked inside the loop.

Reply:
To be honest, I didn't think about the fact that any old garbage could be in the gameboard and notes part of the other `AssignmentDTO`s, so yeah good spot.